tests/integration/testmarketplace.nim

```diff
@@ -179,16 +187,23 @@ marketplacesuite "Marketplace payouts":
     check eventually(slotIdxFilled.isSome, timeout = expiry.int * 1000)
     let slotId = slotId(!clientApi.requestId(id), !slotIdxFilled)
 
+    let purchase = clientApi.getPurchase(id).get
+    check purchase.error == none string
```

Comment:
If we only need the slotSize, then we can use the `datasetSize` proc I suggested above, eg `datasetSize(8, 3, 1)`, instead getting the request from the purchase in the rest API.